chore: update markdown button

deboer-tim opened the pull request:
### What does this PR do?

Buttons generated via markdown weren't matching our regular Button component. I don't love replicating these classes here, but I don't see a simpler way at the moment and this gives it the identical styling. Broke the test html into separate lines so it is testing for the different bits specifically vs the full exact string (and we don't need to duplicate the full set of classes again).

### Screenshot/screencast of this PR

N/A, markdown buttons have minor style differences, e.g. hover is lighter.

### What issues does this PR fix or reference?

N/A

### How to test this PR?

Go to create a Sandbox instance and confirm the 'Login' and 'Create' buttons match.

benoitf commented on packages/renderer/src/lib/markdown/Markdown.spec.ts:
```diff
@@ -48,15 +48,17 @@ describe('Custom button', () => {
     const markdownContent = screen.getByRole('region', { name: 'markdown-content' });
     expect(markdownContent).toBeInTheDocument();
-    expect(markdownContent).toContainHTML('<a class="pf-c-button pf-m-primary" >Name of the button</a>');
+    expect(markdownContent).toContainHTML('Name of the button');
```
I would expect that the test check for the html code of the button, not only a name
It could be a label that is set or just the text so we need to ensure a <button> is generated

deboer-tim replied:
Point taken. I don't love the thought of copying the exact/entire html syntax to the test though. I've added a commit that includes the button and more styling, think this is enough or would you prefer the full html string?

benoitf reviewed (approved):
maybe (I haven't thought when doing the intitial tests) we could have just tried to click on the button and see if we were redirected etc so that it's replaced by a button or something that do what we want

cdrage reviewed (approved):
LGTM!

Tested and Login / Create buttons function correctly.

benoitf commented:
@deboer-tim merge it ?

deboer-tim commented:
We changed the purple color slightly earlier this week, so I'll add a commit later today to update that. Have been trying to build a button/click test as well and I think I'm close.

deboer-tim commented:
Simplified and re-did previous PR:
- Rebased on current main.
- Markdown anchor updated to only replace the PF classes with the same (and current) styling that we apply in Button, and updated tests to match.
- Redid second test to confirm that the markdown was correctly picked up as a link and has the correct attributes.

In other words, I reduced the amount of change but improved the second test. I couldn't find a way to click and catch it with a mock, but at least now we know that it was parsed correctly as an active link with the correct href.
